File: TR-2321-Code-main/BehavePython/BehaveDemo/features/steps/login_steps.py
# from behave import given, when, then
# from selenium import webdriver
# from selenium.webdriver.chrome.options import Options
# from selenium.webdriver.common.by import By
# from selenium.webdriver.support.wait import WebDriverWait
# from selenium.webdriver.support import expected_conditions as EC
#
# BASE_URL = "https://the-internet.herokuapp.com"
#
#
# # ---------- GIVEN STEPS ----------
#
# @given("the application is running")
# def step_application_running(context):
#     """
#     Assert that the application is reachable and responding.
#     We do this by opening the home page and checking for a known element.
#     """
#     options = Options()
#     options.add_argument("--start-maximized")
#
#     context.driver = webdriver.Chrome(options=options)
#     context.wait = WebDriverWait(context.driver, 10)
#
#     context.driver.get(BASE_URL)
#
#     # Assert the application loaded by checking page heading
#     heading = context.wait.until(
#         EC.visibility_of_element_located((By.TAG_NAME, "h1"))
#     )
#     assert heading.text == "Welcome to the-internet"
#
#
# @given("the test database is seeded with users")
# def step_database_seeded(context):
#     # Placeholder for demo purposes
#     # In real systems this might call an API or run a script
#     assert True
#
#
# @given("the user is on the login page")
# def step_user_on_login_page(context):
#     context.driver.get(f"{BASE_URL}/login")
#
#     # Assert login page loaded
#     context.wait.until(
#         EC.visibility_of_element_located((By.ID, "login"))
#     )
#
#
# # ---------- WHEN STEPS ----------
#
# @when('the user enters username "tomsmith"')
# def step_enter_username(context):
#     username_input = context.wait.until(
#         EC.visibility_of_element_located((By.ID, "username"))
#     )
#     username_input.clear()
#     username_input.send_keys("tomsmith")
#
#
# @when('the user enters password "SuperSecretPassword!"')
# def step_enter_password(context):
#     password_input = context.wait.until(
#         EC.visibility_of_element_located((By.ID, "password"))
#     )
#     password_input.clear()
#     password_input.send_keys("SuperSecretPassword!")
#
#
# @when("the user clicks the login button")
# def step_click_login(context):
#     login_button = context.wait.until(
#         EC.element_to_be_clickable((By.CSS_SELECTOR, "button[type='submit']"))
#     )
#     login_button.click()
#
#
# # ---------- THEN STEPS ----------
#
# @then("the user should be redirected to the secure area")
# def step_redirected_to_secure_area(context):
#     context.wait.until(
#         EC.url_contains("/secure")
#     )
#     assert "/secure" in context.driver.current_url
#
#
# @then('the page should display message containing "You logged into a secure area!"')
# def step_verify_success_message(context):
#     flash_message = context.wait.until(
#         EC.visibility_of_element_located((By.ID, "flash"))
#     )
#     assert "You logged into a secure area!" in flash_message.text
#
#     context.driver.quit()

File: TR-2321-Code-main/CucumberJava/CucumberDemo/src/test/java/com/revature/cuc/steps/LoginSteps.java
package com.revature.cuc.steps;

import io.cucumber.java.After;
import io.cucumber.java.Before;
import io.cucumber.java.en.And;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;

import static org.junit.jupiter.api.Assertions.assertTrue;


public class LoginSteps {

//    private WebDriver driver;
//    private final String BASE_URL = "https://the-internet.herokuapp.com/login";
//
//    // ---------- Hooks ----------
//
//    @Before
//    public void setUp() {
//        driver = new ChromeDriver();
//        driver.manage().window().maximize();
//    }
//
//    @After
//    public void tearDown() {
//        if (driver != null) {
//            driver.quit();
//        }
//    }
//
//    // ---------- Background Steps ----------
//
//    @Given("the application is running")
//    public void the_application_is_running() {
//        // Simple health check by opening the login page
//        driver.get(BASE_URL);
//        assertTrue(driver.getTitle().contains("The Internet"));
//    }
//
//    @And("the test database is seeded with users")
//    public void the_test_database_is_seeded_with_users() {
//        // No action required for this demo app
//        // Step exists for readability and future extensibility
//    }
//
//    // ---------- Scenario Steps ----------
//
//    @Given("the user is on the login page")
//    public void the_user_is_on_the_login_page() {
//        driver.get(BASE_URL);
//    }
//
//    @When("the user enters username {string}")
//    public void the_user_enters_username(String username) {
//        driver.findElement(By.id("username")).sendKeys(username);
//    }
//
//    @And("the user enters password {string}")
//    public void the_user_enters_password(String password) {
//        driver.findElement(By.id("password")).sendKeys(password);
//    }
//
//    @And("the user clicks the login button")
//    public void the_user_clicks_the_login_button() {
//        driver.findElement(By.cssSelector("button[type='submit']")).click();
//    }
//
//    @Then("the user should be redirected to the secure area")
//    public void the_user_should_be_redirected_to_the_secure_area() {
//        assertTrue(driver.getCurrentUrl().contains("/secure"));
//    }
//
//    @And("the page should display message containing {string}")
//    public void the_page_should_display_message_containing(String expectedMessage) {
//        String actualMessage = driver.findElement(By.id("flash")).getText();
//        assertTrue(actualMessage.contains(expectedMessage));
//    }
}
